### ISSUE TITLE
Handling TODOs

### ISSUE THREAD

**Issue by flavius** [wirecard/shop-systems-coding-guidelines#1]
* TODO must always be accompanied by ticket number; e.g. you find a bug/issue/improvement, you take a shortcut in the code, but you know how a cleaner approach would look like; naturally, you write a TODO in the code; as part of writing a TODO, you have to create a separate ticket and use that ticket number in your newly written TODO, as a reference to keep things in check
* related to previous, boyscout rule: when you see a TODO with a ticket number in it, check if the referenced ticket has already been tackled, and implement it if so
* related to previous, boyscout rule: when you see a TODO without a ticket number in it, create a ticket for it and update the TODO
* TODOs without ticket numbers are fine during development, as long as they don't reach the code review; useful when you want to get a prototype working without bothering with details, but you need a reminder to address those details before starting a code review

**Comment by flavius** [comment]
We're not allowed to use internal tickets in code, but we could instead:

- create tickets in the internal system and reference code
- create issues on github and reference code

In both cases, the referenced code has to be done via a tree sha id, not via a symbolic branch name. 

In both cases, the disadvantage is that we make it harder for a boy-scout. Why? Because the boy-scout scouts the code, not the tickets. Moreover, a boy-scout might fix an issue without being aware of it (because there is no indication in the code), which means the ticket stays in the backlog untouched, and at some point, someone will have to dig through it, just to figure out that there's nothing to be done => wasted time.
